### PR TITLE
#1389 Filter tweaks

### LINKED ISSUE #1389: Data view filters -- minor tweaks
To be done *after* main feature branch is merged.

Just making a list here and I'll do these together at some point:

- [x] Don't show the "Sort by" filter label, make it an option in the filter menu: "Reset sort order"
- [x] Set a limit on the length of search text in filters and truncate text if it exceeds
- [x] Allow a "default sort column" to be defined (currently just used ID, which should remain the default)
- [x] Changing dataview should reset filters (properly)
- [x] Filter definitions column defaults tweak: https://github.com/openmsupply/conforma-server/issues/902 (Also update table view colums to NOT return filterData by default)
- [x] Also add a "Reset pagination" option in Filter menu
- [x] Update Postman collection
- [x] "Can be blank" option for filter lists (filter parameters)
- [x] Documentation (back-end)

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fix #1389

Bunch of small bug fixes and tweaks as listed in #1389. Requires back-end branch from PR: https://github.com/openmsupply/conforma-server/pull/924

Items addressed by front-end here:

- [x] Don't show the "Sort by" filter label, make it an option in the filter menu: "Reset sort order"
- [x] Set a limit on the length of search text in filters and truncate text if it exceeds
- [x] Changing dataview should reset filters (properly)
- [x] Also add a "Reset pagination" option in Filter menu
- [x] Update Postman collection

See inline comments for more details.

### CODE REVIEW

File: src/containers/DataDisplay/DataViewTable.tsx [hunk 1]
```diff
@@ -52,30 +51,13 @@ const DataViewTable: React.FC = () => {
...
 
-  // Reset filters if navigating here from Menu bar
-  useEffect(() => {
-    if (location?.state?.resetFilters) {
-      setGqlFilter({})
-      setApiQueries({})
-      setSearchText('')
-      setDebounceInput('')
-    }
-  }, [dataViewCode])
```
Comment (CarlosNZ) [inline src/containers/DataDisplay/DataViewTable.tsx:68]: This resetting wasn't quite capturing all cases (it still left "activeFilters", even though they had nothing in them), so moved to "ListFilters" instead. Works better there anyway, as just resetting "activeFilters" resets all of the above as a result.

File: src/containers/DataDisplay/DataViewTable.tsx [hunk 2]
```diff
@@ -52,30 +51,13 @@ const DataViewTable: React.FC = () => {
     filtersReady: !filtersLoading,
   })
 
-  const [currentSortColumn, setCurrentSortColumn] = useState<SortColumn>(null)
```
Comment (CarlosNZ) [inline src/containers/DataDisplay/DataViewTable.tsx:55]: Can get rid of quite a lot as a result of not showing the currently sorted column in the filter bar 👍🏼  (Most of the removed content)

File: src/containers/DataDisplay/Filters/DataViewSearchableList.tsx [hunk 3]
```diff
@@ -32,13 +32,13 @@ export const DataViewSearchableList: React.FC<SearchableListProps> = ({
   const [moreResults, setMoreResults] = useState(false)
   const [debounceOutput, setDebounceInput] = useDebounce(searchText)
 
-  const { column, code, searchFields, delimiter } = options
+  const { column, code, ...otherOptions } = options
```
Comment (CarlosNZ) [inline src/containers/DataDisplay/Filters/DataViewSearchableList.tsx:35]: These properties (plus the `includeNull` prop) are not actually used in the front-end, but passed back unchanged to server with search queries, so no need to unpack them. And it means if we add more "filter_parameters" we won't need to deal with them here (unless they're relevant to front-end)

File: src/containers/Dev/Snapshots.tsx [hunk 4]
```diff
@@ -64,7 +64,11 @@ const Snapshots: React.FC = () => {
         url: getServerUrl('snapshot', { action: 'use', name }),
       })
 
-      if (resultJson.success) return setIsLoading(false)
+      if (resultJson.success) {
+        setIsLoading(false)
+        location.reload(true)
+        return
+      }
```
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:71]: Not related to this PR. This just makes the whole app reload after a snapshot is loaded, which is much nicer.

File: src/containers/List/ListFilters/ListFilters.tsx [hunk 5]
```diff
@@ -49,6 +49,12 @@ const ListFilters: React.FC<{
 
   // Add filters from URL to activeFilters, filters may not have criteria/options yet thus we have to use activeFilter state variable
   useEffect(() => {
+    // Reset active filters if coming from Database menu selection
+    if (location?.state?.resetFilters) {
+      setActiveFilters([])
+      return
+    }
```
Comment (CarlosNZ) [inline src/containers/List/ListFilters/ListFilters.tsx:56]: As mentioned above, moved from "DataViewTable"

File: src/containers/List/ListFilters/ListFilters.tsx [hunk 6]
```diff
@@ -110,10 +116,27 @@ const ListFilters: React.FC<{
...
+          )}
+          {(query.page || query.perPage) && (
+            <Dropdown.Item
+              key="clear-pagination"
+              onClick={() => updateQuery({ page: null, perPage: null })}
+            >
+              <Icon size="small" color="grey" name="delete" />
+              {strings.FILTER_RESET_PAGINATION}
+            </Dropdown.Item>
+          )}
```
Comment (CarlosNZ) [inline src/containers/List/ListFilters/ListFilters.tsx:139]: This is the "Reset Filters", "Reset Sort" and "Reset pagination" controls. I've made them all only appear if there's actually something to reset. Does that seem okay, or do you reckon it's better to have them in the menu all the time regardless?

File: src/utils/hooks/useDataViews.tsx [hunk 7]
```diff
@@ -182,26 +182,22 @@ const buildFilterDefinitions = (
...
-          code,
-          dataType,
-          showFilterList,
-          searchFields,
-          delimiter,
-          booleanMapping,
-          nullString,
-        },
-      }
+  filterDefinitions.forEach(({ column, title, dataType, showFilterList, ...otherOptions }) => {
```
Comment (CarlosNZ) [inline src/utils/hooks/useDataViews.tsx:185]: As above, just passing through `...otherOptions` without destructuring them

File: src/utils/defaultLanguageStrings.ts [hunk 8]
```diff
@@ -133,7 +133,8 @@ export default {
   FILTER_START_TYPING: 'Start typing...',
   FILTER_SEARCH_LIST: 'Search list...',
   FILTER_SORT_BY: 'Sort by',
-  FILTER_RESET_SORT: 'Reset sort',
+  FILTER_RESET_SORT: 'Reset column sort',
+  FILTER_RESET_PAGINATION: 'Reset pagination',
```
Comment (CarlosNZ) [inline src/utils/defaultLanguageStrings.ts:137]: @nmadruga You might want to pre-translate these into the Angola localisations file so they're already there when this PR is merged.

File: src/containers/Dev/Snapshots.tsx (hunk 4 above)
Comment (nmadruga) [reply src/containers/Dev/Snapshots.tsx:71]: Oh! Cool, I'll have a look on how this is looking : )

**Review by nmadruga — APPROVED** [review]
👍 I liked the improvements a lot

**Pull request merged**